Utils/azuretools-core/src/com/microsoft/azuretools/utils/WebAppUtils.java

```diff
@@ -146,6 +147,45 @@ public static int deployArtifact(String artifactName, String artifactPath, Publi
         return uploadingTryCount;
     }
 
+    public static int deployArtifactForJavaSE(String artifactPath, PublishingProfile pp, IProgressIndicator indicator) throws IOException {
+        File file = new File(artifactPath);
+        if (!file.exists()) {
+            throw new FileNotFoundException(String.format(NO_TARGET_FILE, artifactPath));
+        }
+        FTPClient ftp = null;
+        int uploadingTryCount;
+        try (InputStream input = new FileInputStream(artifactPath)){
+            if (indicator != null) {
+                indicator.setText("Connecting to FTP server...");
+            }
+            ftp = getFtpConnection(pp);
+            if (indicator != null) {
+                indicator.setText("Uploading the application...");
+            }
+            uploadingTryCount = uploadFileToFtp(ftp, ftpRootPath + JAVASE_ROOT + "." + TYPE_JAR, input, indicator);
+            if (indicator != null) {
+                indicator.setText("Logging out of FTP server...");
+            }
+            ftp.logout();
+        } finally {
+            if (ftp != null && ftp.isConnected()) {
+                ftp.disconnect();
+            }
+        }
+        return uploadingTryCount;
+    }
+
+    private static void ensureWebAppsFolderExist(FTPClient ftp) {
+        try {
+            ftp.getStatus(ftpWebAppsPath);
+            if (!FTPReply.isPositiveCompletion(ftp.getReplyCode())) {
+                ftp.makeDirectory(ftpWebAppsPath);
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
why swallow  all error here?

Reply:
The method ensureWebAppsFolderExist can fail. So just swallow the exception.

Reply:
Try to create ftp directory 3 time, if still fails, throw exception